backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java

```diff
@@ -116,16 +116,39 @@ public DatasetVersionServiceImpl(
           new ArrayList<>(datasetIdSet).get(0));
       accessibleDatasetVersionIds.addAll(requestedDatasetVersionIds);
     } else {
-      allowedDatasetIds =
+      allowedDatasetVersionIds =
           roleService.getSelfAllowedResources(
               ModelDBServiceResourceTypes.DATASET, modelDBServiceActions);
       // Validate if current user has access to the entity or not
-      allowedDatasetIds.retainAll(requestedDatasetVersionIds);
-      accessibleDatasetVersionIds.addAll(allowedDatasetIds);
+      allowedDatasetVersionIds.retainAll(datasetIdSet);
+      for (Map.Entry<String, String> entry : datasetIdDatasetVersionIdMap.entrySet()) {
+        if (allowedDatasetVersionIds.contains(entry.getValue())) {
+          accessibleDatasetVersionIds.add(entry.getKey());
+        }
+      }
     }
     return accessibleDatasetVersionIds;
   }
 
+  public List<String> getAccessibleDatasetVersionIDsByAction(
+      List<String> requestedDatasetVersionIds, ModelDBServiceActions modelDBServiceActions) {
+    // Validate if current user has access to the entity or not
+    if (requestedDatasetVersionIds.size() == 1) {
+      roleService.isSelfAllowed(
```

Comment:
let us say a user has delete privilages on Dataset and deos not have a role binding at dataset version . how will this call to isSelfAllowed  working?

Reply:
resolved conversation.